Source/WebCore/page/scrolling/ScrollingCoordinatorTypes.cpp

```diff
@@ -43,6 +43,14 @@ void RequestedScrollData::merge(RequestedScrollData&& other)
             other.requestedDataBeforeAnimatedScroll = requestedDataBeforeAnimatedScroll;
             break;
         }
+    } else if (requestType == ScrollRequestType::DeltaUpdate && other.requestType == ScrollRequestType::DeltaUpdate) {
+        other.scrollPositionOrDelta = std::get<FloatSize>(scrollPositionOrDelta) + std::get<FloatSize>(other.scrollPositionOrDelta);
+    } else if (requestType == ScrollRequestType::DeltaUpdate || other.requestType == ScrollRequestType::DeltaUpdate) {
+        other.requestType = ScrollRequestType::PositionUpdate;
+        if (requestType == ScrollRequestType::DeltaUpdate)
+            other.scrollPositionOrDelta = std::get<FloatPoint>(other.scrollPositionOrDelta) + std::get<FloatSize>(scrollPositionOrDelta);
+        else
+            other.scrollPositionOrDelta = std::get<FloatPoint>(scrollPositionOrDelta) + std::get<FloatSize>(other.scrollPositionOrDelta);
```

Comment:
What if one of the two request types is `CancelAnimatedScroll` and `animated == ScrollIsAnimated::Yes`? I think we can still get in here in that case, and we shouldn't be inspecting `scrollPositionOrDelta` on it.

Reply:
Right updated to only consider these cases for when the new request is a delta update, and the current request is a delta update or position update.